tests/test_conversion.py

```diff
@@ -720,3 +720,28 @@ def test_intptr_construction():
         with pytest.raises(OverflowError):
             UIntPtr(v)
 
+def test_explicit_conversion():
+    from System import (
+        Int64, UInt64, Int32, UInt32, Int16, UInt16, Byte, SByte, Boolean
+    )
+    from System import Double, Single
+
+    assert int(Boolean(False)) == 0
+    assert int(Boolean(True)) == 1
+
+    for t in [UInt64, UInt32, UInt16, Byte]:
+        assert int(t(127)) == 127
+        assert float(t(127)) == 127.0
+
+    for t in [Int64, Int32, Int16, SByte]:
+        assert int(t(127)) == 127
+        assert int(t(-127)) == -127
+        assert float(t(127)) == 127.0
+        assert float(t(-127)) == -127.0
+
+    assert int(Int64.MaxValue) == 2**63 - 1
+    assert int(Int64.MinValue) == -2**63
+    assert int(UInt64.MaxValue) == 2**64 - 1
```

Comment:
I believe this does not invoke the new code. Only explicit construction.

Reply:
No, explicit construction is the other dierction. This does call the new code.

Reply:
Have you actually run it? Accessing `Int64.MaxValue` is reading a field, and upon reading any fields or properties I believe conversions happen (like `Int32` -> `int` or `String` -> `str`), so `Int64.MaxValue` is already returning `int` type to Python. And then `int(int value)` does nothing.

Reply:
Oooh, yes, sorry. I have a follow-up PR (#1907), I'll add this. Maybe we should just do away with the implicit conversions altogether :/

Reply:
Fixed in https://github.com/pythonnet/pythonnet/pull/1907/commits/c7068a522a8c84584515a0bfc2c97c2bfc5e88b8 (I verified it in IPython this time ;))

Reply:
That would involve quite a bit of work, and may be slightly unexpected. Most Python code expects `int` and even slight deviation from its behavior will cause unexpected issues.

For example, I am not sure what behavior `int` + `System.Byte` has. E.g. does it convert `Byte` to `int` and performs full addition, or converts `int` to one of .NET integer types, and subsequently wraps on overflow.

Some code will even have explicit `isinstance` checks.